src/test/engineeffectsdelay_test.cpp

```diff
@@ -31,14 +30,18 @@ class EngineEffectsDelayTest : public MixxxTest {
             const std::span<const CSAMPLE> referenceBuffer) {
         EXPECT_EQ(buffer.size(), referenceBuffer.size());
 
-        std::transform(buffer.begin(),
-                buffer.end(),
-                referenceBuffer.begin(),
-                buffer.begin(),
-                [&](const auto& original, const auto& reference) {
-                    EXPECT_FLOAT_EQ(original, reference);
-                    return original;
-                });
+        std::span<CSAMPLE>::iterator pBufferIterator =
+                buffer.begin();
+        std::span<const CSAMPLE>::iterator pReferenceBufferIterator =
+                referenceBuffer.begin();
+
+        while (pBufferIterator != buffer.end() &&
+                pReferenceBufferIterator != referenceBuffer.end()) {
+            EXPECT_FLOAT_EQ(*pBufferIterator, *pReferenceBufferIterator);
+
+            ++pBufferIterator;
+            ++pReferenceBufferIterator;
+        }
```

Comment:
Okay, so can I use it instead and avoid using iterators in this special case? IMO more readable solution.

Reply:
Yes, but only because we asserted that the sizes were the same. Alternatively you can use the bounds-checked accessor `span.at(i)` in the loop.

Reply:
Maybe not so clear, as I expected. The data type has to be explicitly written (can't use auto), to avoid casting, so the solution looks like this:

```
for (std::span<CSAMPLE>::size_type i = 0; i < buffer.size(); i++) {
    EXPECT_FLOAT_EQ(buffer[i], referenceBuffer[i]);
}
```

What would you prefer?

Reply:
Are you sure, it works for me...
https://compiler-explorer.com/z/q77Eaa8jq

Reply:
Even works on gcc 10

Reply:
Yeah, it works, that's right. The compiler shows this warning only:
```
/home/david/Documents/CodingFiles/GitWorkspace/Mixxx/mixxx/src/test/engineeffectsdelay_test.cpp: In member function ‘void {anonymous}::EngineEffectsDelayTest::AssertIdenticalBufferEquals(std::span<float>, std::span<const float>)’:
/home/david/Documents/CodingFiles/GitWorkspace/Mixxx/mixxx/src/test/engineeffectsdelay_test.cpp:33:28: warning: comparison of integer expressions of different signedness: ‘int’ and ‘std::span<float>::size_type’ {aka ‘long unsigned int’} [-Wsign-compare]
   33 |         for (auto i = 0; i < buffer.size(); i++) {
      |                          ~~^~~~~~~~~~~~~~~
```

but I suppose that the clazy test on CI will fail after updating on Ubuntu 22 still.